### ISSUE TITLE
Sum type parsing

### ISSUE THREAD

**Issue by MaxGabriel** [Gabriel439/Haskell-Optparse-Generic-Library#50]
Hi, I wanted to use a sum type in my command line parsing, and I came up with this:

```haskell
data FrontendLanguage = Typescript | Swift deriving (Eq, Show, Read, Generic)

instance ParseField FrontendLanguage
instance ParseRecord FrontendLanguage
instance ParseFields FrontendLanguage

data Tasks
    = GenerateRoutes 
      { language :: FrontendLanguage <?> "Options are: Typescript, Swift"
      , frontendRepo :: FilePath <?> "Path to the root of the frontend repo"
      }
    deriving (Generic, Show)

instance ParseRecord Tasks
```

It works fine in the success case, though there isn't a specific error message if you provide an invalid sum type, just the generic help text.

Is this the best way to do this? I couldn't find anything in the documentation about this (could have missed it, though). If it is the best way, I'll submit a PR to document this use-case in the Haddocks if you'd like.

**Comment by Gabriella439** [comment]
Yeah, that's about the best you can do within this library.  You can do better if you use the underlying `optparse-applicative` library to define the `ParseFields` instance for `FrontendLanguage` because then you can define it to be something like `( --typescript | --swift )`
